opus/application/apps/cart/views.py

```diff
@@ -693,22 +693,23 @@ def _get_cart_count(session_id):
 #
 ################################################################################
 
-def _add_to_cart_table(opus_id_list, session_id, api_code):
+def _add_to_cart_table(opus_id, session_id, api_code):
     "Add OPUS_IDs to the cart table."
     cursor = connection.cursor()
-    if not isinstance(opus_id_list, (list, tuple)):
-        opus_id_list = [opus_id_list]
-    res = (ObsGeneral.objects.filter(opus_id__in=opus_id_list)
+    res = (ObsGeneral.objects.filter(opus_id__exact=opus_id)
            .values_list('opus_id', 'id'))
-    if len(opus_id_list) != len(res):
-        return 'opusid not found'
+    if len(res) != 1:
+        return (f'Internal Error: OPUS ID {opus_id} not found; nothing added '
+                +'to cart')
 
     # Note that this doesn't handle the case where some or all of the opus_ids
     # are already in the cart. It's difficult to handle this well, would slow
     # down the API, and should never happen when using the UI anyway.
     num_selections = _get_cart_count(session_id)
-    if num_selections+len(opus_id_list) > settings.MAX_SELECTIONS_ALLOWED:
-        return 'Too many observations in cart'
+    if num_selections >= settings.MAX_SELECTIONS_ALLOWED:
+        return (f'Your request to add OPUS ID {opus_id} to the cart failed '
```

Comment:
is this going to post a message w/all the opus IDs that did not get added?  I"m not clear that OPUS_ID {opus_id} is needed/useful.

Reply:
Add only allows one opus id so this message just displays one opus id. Since you can have multiple add requests outstanding due to network delay, I thought it would be useful to tell people which one failed. Otherwise you could get a series of modals that all say exactly the same thing, which would probably be annoying.